### PR TITLE
Replace utils.ExitOnError->exit.OnErrorWithMessage

### PR DESCRIPTION
**Pull request by Jaanki**
Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: internal/exit/exit.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func WithMessage(message string) {
 }
 
 // OnErrorWithMessage will print the message and quit the program with an error code.
-func OnErrorWithMessage(err error, message string) {
+func OnErrorWithMessage(message string, err error) {
```
Comment (tpantelis) [inline internal/exit/exit.go]: Why was the signature changed? I prefer the previous ordering which is consistent with `errors.Wrap`, `errors.WithMessage` et al.

File: internal/restconfig/restconfig.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ package restconfig
 
 import (
 	"fmt"
+	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
```
Comment (tpantelis) [inline internal/restconfig/restconfig.go]: `utils` is no longer referenced.

**Review by tpantelis — CHANGES_REQUESTED** [review]
We should be able to remove `utils.ExitOnError`.

File: internal/exit/exit.go (hunk 1 above)
Comment (Jaanki) [reply internal/exit/exit.go]: It was changed to match the output message `custom message: err`.

File: internal/restconfig/restconfig.go (hunk 2 above)
Comment (Jaanki) [reply internal/restconfig/restconfig.go]: L302 still uses `utils.GetSubmarinerResource(config)`

File: internal/exit/exit.go (hunk 1 above)
Comment (tpantelis) [reply internal/exit/exit.go]: But the runtime output produced isn't related to the param ordering. The param ordering should reflect the function name. Please keep the ordering as it was for consistency.

**Review by tpantelis — APPROVED** [review]

**Review by skitt — APPROVED** [review]
This is much clearer!

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**